src/couch/src/couch_db.erl

```diff
@@ -1752,11 +1752,13 @@ set_namespace_range(Options, NS) ->
         true ->
             SK = select_gt(SK0, <<NS/binary, "/">>),
             EK = select_lt(EK0, <<NS/binary, "0">>),
-            [{start_key, SK}, {EKType, EK}];
+            [{dir, proplists:get_value(dir, Options, fwd)},
+                {start_key, SK}, {EKType, EK}];
         false ->
             SK = select_lt(SK0, <<NS/binary, "0">>),
             EK = select_gt(EK0, <<NS/binary, "/">>),
-            [{dir, rev}, {start_key, SK}, {EKType, EK}]
+            [{dir, proplists:get_value(dir, Options, rev)},
```

Comment:
The default is fwd, not rev.

Reply:
I should note that this is a long standing API oddity that we've gone back and forth on. I.e., do we try and help the user by swapping keys when we can detect that they'd have an empty result or not. Unless I'm forgetting a change we've always landed on the "don't to it" as depending on how you read the request it violates what they ask for.